includes/ThanksCache.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace MediaWiki\Extension\Thanks;
+
+use Config;
+use RequestContext;
+use Wikimedia\Rdbms\ILoadBalancer;
+use Wikimedia\Rdbms\SelectQueryBuilder;
+
+/**
+ * A service class that caches the list of IDs (revisions, logs) a user has thanked for.
+ * It uses a session as a backend.
+ */
+class ThanksCache {
+	private const THANKED_IDS_SESSION_KEY_PATTERN = "thanks-thanked-ids-%d";
+	/** @var ILoadBalancer */
+	private $lb;
+	/** @var Config */
+	private $config;
+
+	public function __construct( ILoadBalancer $lb, Config $config ) {
+		$this->lb = $lb;
+		$this->config = $config;
+	}
+
+	/**
+	 * Gets a cached list of entires for which a user has thanked.
+	 * Entires are refetched on cache miss.
+	 *
+	 * Each entry is in the same format as saved by the store (i.e. a string formatted as "$type-$id").
+	 * @param int $thankerActorId - actor ID of the user who thanked.
+	 * @return array An array of strings formatted as "$type-$id"
+	 */
+	public function getUserThanks( int $thankerActorId ): array {
+		$dbr = $this->lb->getConnection( DB_REPLICA );
+		$session = RequestContext::getMain()->getRequest()->getSession();
```

Comment:
Let's pass in an `IContextSource` as a method parameter instead so that it can facilitate testing later. It could then be naturally sourced from the local context in e.g. `ApiCoreThank`.